### PR TITLE
Update Java lexer

### LINKED ISSUE #872: Java text blocks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What problem does this feature solve?

Chroma should support text blocks which got added in Java 15:

```java
public static void main(String[] args) {
    System.out.println("""
            Hello, world!
            This is a multi-line string!
            It can also contain "quotes" and 'apostrophes' without breaking.
            """);
}
```

Chroma Playground (v2.9.1):

image

### What feature do you propose?

The textblock should render as such:

image

Source: IntelliJ

### PR DESCRIPTION
**Pull request by otbutz**
Fixes #872 

Generated from pygments v2.16.1.

```sh
python3 _tools/pygments2chroma_xml.py pygments.lexers.jvm.JavaLexer > lexers/embedded/java.xml
```

grafik

### CODE REVIEW

**Comment by otbutz** [comment]
Fixes #872

File: lexers/embedded/java.xml [hunk 1]
```diff
@@ -5,116 +5,188 @@
     <filename>*.java</filename>
     <mime_type>text/x-java</mime_type>
     <dot_all>true</dot_all>
-    <ensure_nl>true</ensure_nl>
```
Comment (otbutz) [inline lexers/embedded/java.xml:8]: This got dropped while regenerating the file.

**Comment by otbutz** [comment]
@alecthomas the test errors for cql and markdown look unrelated to my changes but follow the same trend as the changes in the newly generated Java lexer definition.

* `Operator` -> `Punctuation`
* `Text` -> `TextWhitespace`

**Comment by otbutz** [comment]
@alecthomas should I regenerate the expected files for these formats?

**Comment by alecthomas** [comment]
Yes. They're failing because they embed Java.

**Comment by otbutz** [comment]
Done :)

**Comment by alecthomas** [comment]
Thanks!

**Comment by otbutz** [comment]
Any timeline for the next release? Would love to get the change into the upcoming gitea relase :wink:

**Comment by alecthomas** [comment]
Sure thing, tagged 2.10.0

**Comment by otbutz** [comment]
@alecthomas could you bump the playground version to v2.10.0?

**Pull request merged**